src/main/xerus/monstercat/api/Covers.kt

```diff
@@ -16,13 +16,16 @@ object Covers {
 	private fun coverCacheFile(coverUrl: String, size: Int): File {
 		coverCacheDir.mkdirs()
 		val newFile = coverCacheDir.resolve(coverUrl.substringBeforeLast('/').substringAfterLast('/').replaceIllegalFileChars())
-		return coverCacheDir.resolve("${newFile.nameWithoutExtension}x$size.${newFile.extension}")
+		return coverCacheDir.resolve("${newFile.nameWithoutExtension}x$size.jpg") // FIXME : Do not hardcode extension
 	}
 		
 	/** Returns an Image of the cover in the requested size using caching.
 	 * @param size the size of the Image that is returned - the image file will always be 64x64 */
-	fun getThumbnailImage(coverUrl: String, size: Number = 64, invalidate: Boolean = false): Image =
+	fun getThumbnailImage(coverUrl: String, size: Number = 64, invalidate: Boolean = false) = try {
 		getCover(coverUrl, 64, invalidate).use { createImage(it, size) }
+	} catch (e: Exception) {
+		null
+	}
```

Comment:
Why swallow the exception here?

And if you have to, please document that.

Reply:
Indeed, there should be an error logging.

Reply:
Why does this function have to catch the error at all, why not propagate it? And you should almost never catch a blanket-Exception - this might swallow critical unexpected errors too.

Reply:
Better handling of exceptions with 81f8555b103b1747f228d7f914925b37b2d69c4a